### ISSUE TITLE
kde-plasma/* dependencies

### ISSUE THREAD

**Issue by barolgin** [gentoo/kde#367]
`kde-plasma/breeze-9999` depends on `kde-frameworks/plasma-9999` because of files `KF5PlasmaConfig.cmake` and  `kf5plasma-config.cmake`. Maybe the whole kde-plasma/* section has these dependencies. But while emerging `kde-plasma/breeze-9999` package `kde-frameworks/plasma-9999` is not intended for emerge.

**Comment by kensington** [comment]
We hear about it every time we change this behaviour. :smile: 

We have two options to handle this sort of situation - make live packages always require live dependences, and to have no special dependency version handling.

After switching backwards and forwards a few times, we've settled on the current situation (no special version handling) primarily because of one reason - most of the time a live ebuild doesn't actually require live dependencies.

To do otherwise would result in the entire KF5 deptree being rebuilt as live, which is a lot of packages and often unnecessary.

**Comment by johu** [comment]
:-1: While a release cycle plasma always requires unreleased frameworks. I am against to allow not live.

**Comment by kensington** [comment]
It's not always required. In fact, I was able to build breeze-9999 against plasma-5.13.0 just fine.

@barolgin please add failing build log

**Comment by barolgin** [comment]
Sorry, I wrote OP incorrectly, `breeze-9999` has no dependecy from plasma at all (regardless of specific version). Yes `breeze-9999` builds with `plasma-5.13.0` fine. But in clean system, which has not plasma installed:
```
emerge --search "%@kde-frameworks/plasma"                                                                                                                                                                                                                
[ Results for search key : %@kde-frameworks/plasma ]
Searching...

*  kde-frameworks/plasma
      Latest version available: 9999
      Latest version installed: [ Not Installed ]
      Size of files: 0 KiB
      Homepage:      https://www.kde.org/
      Description:   Plasma framework
      License:       LGPL-2+
```

while trying to emerge breeze got this dependence list:
```
emerge -avD kde-plasma/breeze

 * IMPORTANT: 1 news items need reading for repository 'gentoo'.
 * Use eselect news read to view new items.


These are the packages that would be merged, in order:

Calculating dependencies... done!
[ebuild  N     ] media-libs/libuninameslist-20091231::gentoo  0 KiB
[ebuild  NS    ] app-text/docbook-xml-dtd-4.5-r1:4.5::gentoo [4.1.2-r6:4.1.2::gentoo, 4.2-r2:4.2::gentoo, 4.3-r1:4.3::gentoo, 4.4-r2:4.4::gentoo] 0 KiB
[ebuild  N     ] x11-libs/libxkbui-1.0.2-r1::gentoo  USE="-static-libs" 0 KiB
[ebuild  N     ] media-gfx/fontforge-20150430::gentoo  USE="X cairo gif gtk jpeg png python readline svg tiff unicode -truetype-debugger" PYTHON_SINGLE_TARGET="python2_7 -python3_3 -python3_4" PYTHON_TARGETS="python2_7 python3_4 -python3_3" 0 KiB
[ebuild  N    *] kde-frameworks/ki18n-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kwidgetsaddons-9999:5/9999::kde  USE="nls -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kdbusaddons-9999:5/9999::kde  USE="X nls -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kcrash-9999:5/9999::kde  USE="X nls -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kitemviews-9999:5/9999::kde  USE="nls -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/karchive-9999:5/9999::kde  USE="bzip2 lzma -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kguiaddons-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-plasma/kdecoration-9999:5::kde  USE="-debug" 0 KiB
[ebuild  N    *] kde-frameworks/kitemmodels-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/sonnet-9999:5/9999::kde  USE="aspell nls -debug -doc -hunspell {-test}" 0 KiB
[ebuild  N    *] kde-plasma/oxygen-fonts-9999:5::kde  USE="X" 0 KiB
[ebuild  N    *] kde-frameworks/kdoctools-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kcompletion-9999:5/9999::kde  USE="nls -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kjobwidgets-9999:5/9999::kde  USE="X nls -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kglobalaccel-9999:5/9999::kde  USE="nls -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kpty-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kservice-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kpackage-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kdesu-9999:5/9999::kde  USE="X -debug -doc" 0 KiB
[ebuild  N    *] kde-frameworks/kemoticons-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N     ] sys-auth/polkit-qt-0.112.0-r1::gentoo  USE="qt5 -debug -examples -qt4" 0 KiB
[ebuild  N    *] kde-frameworks/kunitconversion-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/solid-9999:5/9999::kde  USE="nls -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kauth-9999:5/9999::kde  USE="nls policykit -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kconfigwidgets-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kiconthemes-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-plasma/polkit-kde-agent-9999:5::kde  USE="-debug" 0 KiB
[ebuild  N    *] kde-frameworks/ktextwidgets-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kwallet-9999:5/9999::kde  USE="-debug -doc (-gpg) {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kxmlgui-9999:5/9999::kde  USE="-attica -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kbookmarks-9999:5/9999::kde  USE="nls -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kio-9999:5/9999::kde  USE="X acl kerberos -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kinit-9999:5/9999::kde  USE="caps -debug -doc" 0 KiB
[ebuild  N    *] kde-frameworks/kded-9999:5/9999::kde  USE="-debug -doc" 0 KiB
[ebuild  N    *] kde-frameworks/kparts-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/frameworkintegration-9999:5/9999::kde  USE="X -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kdeclarative-9999:5/9999::kde  USE="-debug -doc" 0 KiB
[ebuild  N    *] kde-frameworks/kcmutils-9999:5/9999::kde  USE="-debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-frameworks/kdewebkit-9999:5/9999::kde  USE="-debug -doc" 0 KiB
[ebuild  N    *] kde-frameworks/kdesignerplugin-9999:5/9999::kde  USE="nls webkit -debug -designer -doc" 0 KiB
[ebuild  N    *] kde-frameworks/kdelibs4support-9999:5/9999::kde  USE="X -debug -doc {-test}" 0 KiB
[ebuild  N    *] kde-plasma/kde-cli-tools-9999:5::kde  USE="X handbook -debug {-test}" 0 KiB
[ebuild  N    *] kde-plasma/breeze-9999:5::kde  USE="-debug -qt4" 0 KiB

Total: 47 packages (46 new, 1 in new slot), Size of downloads: 0 KiB
```

without any version of plasma.
So while emerging breeze got cmake error:
```
  ...
  CMake Error at CMakeLists.txt:34 (find_package):
  Could not find a package configuration file provided by "KF5Plasma" with
  any of the following names:

    KF5PlasmaConfig.cmake
    kf5plasma-config.cmake

  Add the installation prefix of "KF5Plasma" to CMAKE_PREFIX_PATH or set
  "KF5Plasma_DIR" to a directory containing one of the above files.  If
  "KF5Plasma" provides a separate development package or SDK, be sure it has
  been installed.
  ```

**Comment by kensington** [comment]
Ah, so it's a missing dependency.

Thanks, fixed in 4ea81607805fcb0888f1a9ce03a7151d542feac1.
